### ISSUE TITLE
TANF Section 1 validation clean-up

### ISSUE THREAD

**Issue by ADPennington** [raft-tech/TANF-app#2681]
**Description:**

After #2369 is implemented, some of the validation checks for TANF Section 1 data need updates, either due to on-going revisions OFA is making based on STT feedback to the new coding instructions or because the pre-specified checks have not been implemented. These include:

- ~~`DATE_OF_BIRTH` in the T2 and T3 records are stored and validated as integers instead of date (YYYYMMDD)~~
    - Moved to #2746
- `WORK_PART_STATUS` values are `01,02,05,07,09,15,16,17,18,99`
- `if item 48 == 11 and AGE < 19, then item 38 != 1`
- `FIPS_CODE` still needs to be stored in T1  from the header record  (`state_fips`) 
- `FUNDING_STREAM` value options will revert back to only having 2 options. OFA has decided not to use value `3`. 
- ~~`GENDER` should allow for space-fill (T2 and T3)~~
- `OTHER_AMOUNT` and `TRANSITION_SERVICES_AMOUNT` should be excluded from sum amount cat 3 validator. 
- `DISPOSITION` only has one possible value now. anything that isnt `1` should be flagged as out of range. 

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [ ] validation checks noted above are accounted for
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
_Create a list of granular, specific work items that must be completed to deliver the desired outcomes of this issue_ 

- [ ] _Task 1_
- [ ] _Task 2_
- [ ] _Task 3_
- [ ] Run Testing Checklist and confirm all tests pass

**Notes:**
_Add additional useful information, such as related issues and functionality that isn't covered by this specific issue, and other considerations that will be helpful for anyone reading this_ 

- @ADPennington to create an ssp section 1 enhancement ticket
- _Note 2_
- _Note 3_

**Supporting Documentation:**
_Please include any relevant log snippets/files/screen shots_

- [_TANF Sec 1 WorkBook_](https://gorafttech.sharepoint.com/sites/TDRSResearchDesign/Shared%20Documents/Forms/AllItems.aspx?id=%2Fsites%2FTDRSResearchDesign%2FShared%20Documents%2FDev%2Fdev%2FParsing%2Fdata%2Ftanf%2FSection1&viewid=981949f0%2D98e1%2D4ace%2D8da8%2D436aca592a46)
- _Doc 2_

**Open Questions:**
_Please include any questions or decisions that must be made before beginning work or to confidently call this issue complete_ 

- _Open Question 1_
- _Open Question 2_

**Comment by ADPennington** [comment]
in scenarios where fam affil == 3: caretaker, some states (e.g. SD) will space-fill many of the T2 data values. this is currently being flagged as errors in prod @lfrohlich @ysong001 @ttran-hub

**Comment by ysong001** [comment]
that would be problem, as states did use space-fill before

**Comment by ADPennington** [comment]
@ysong001 TDRS and TDP suggests that maybe only a handful of states are doing this. to be sure, we'll need to look into the T2 data elements in TDRS that have space-filled values for items like: `PARENT_MINOR_CHILD`, `NEEDS_PREGNANT_WOMAN`,`EDUCATION_LEVEL`, etc.

**Comment by ADPennington** [comment]
### Tracking STTs with space-filled values for Section 1 data elements (using FY23Q1 data)

#### `T1` records
![county_spaces_bystate](https://github.com/raft-tech/TANF-app/assets/63075587/61affd59-9c38-4665-82c3-da7bbc254d94)

#### `T2` records
![race_spaces_bystate](https://github.com/raft-tech/TANF-app/assets/63075587/edf84dbb-0395-49b9-ba3c-33963e4503c8)
![mar_spaces_bystate](https://github.com/raft-tech/TANF-app/assets/63075587/f38af14f-097c-4060-b6cc-8489b64c596b)
![tl_spaces_bystate](https://github.com/raft-tech/TANF-app/assets/63075587/bfd81ba4-1ac3-449f-a521-a45f88d43501)
![unsub_spaces_bystate](https://github.com/raft-tech/TANF-app/assets/63075587/c9920670-df2f-4667-9423-180f8f179668)
![ojt_spaces_bystate](https://github.com/raft-tech/TANF-app/assets/63075587/6ac5782c-1633-473c-ad55-b27a55be380a)
![com_spaces_bystate](https://github.com/raft-tech/TANF-app/assets/63075587/c7c0053b-1321-437f-b102-2a5d52a75f44)
![js_spaces_bystate](https://github.com/raft-tech/TANF-app/assets/63075587/3261fd85-075b-427a-86ff-3e83d8d32d4c)
![sa_spaces_bystate](https://github.com/raft-tech/TANF-app/assets/63075587/f90fc8d1-d8f1-4a97-a7f8-0b0f0005ff0f)
![deem_spaces_bystate](https://github.com/raft-tech/TANF-app/assets/63075587/b37252bf-b7dd-45b7-8b9e-c15e37a70cb3)

#### `T3` records

![t3_spaces_bystate](https://github.com/raft-tech/TANF-app/assets/63075587/65bbcde8-692c-4353-9c08-21fef3a8432f)

**Comment by ysong001** [comment]
I was wondering if the non-required person can use blanks, CA asked that they use blanks for the data elements they don't report

**Comment by ADPennington** [comment]
given the [results above](https://github.com/raft-tech/TANF-app/issues/2681#issuecomment-1702820720) and given our coding instructions guidance (referenced below) we should discuss how to handle. 

**One reminder is that, currently, while these issues will appear in the feedback report, the data are still being added to the db.**

> Each item needs to hold a certain number of spaces in the flat file. If your system does not automatically hold the spaces, see the file layout for the number of spaces (length) that need to be held. If an element is not applicable or no longer in use, enter the number of digits (zeros or blanks)

**Comment by ADPennington** [comment]
@ysong001 @lfrohlich i'm creating a new ticket for space-filling.

**Comment by ADPennington** [comment]
@jtimpe this ticket has been refined.

**Comment by ADPennington** [comment]
tanf section 1 workbook: https://gorafttech.sharepoint.com/sites/TDRSResearchDesign/Shared%20Documents/Forms/AllItems.aspx?id=%2Fsites%2FTDRSResearchDesign%2FShared%20Documents%2FDev%2Fdev%2FParsing%2Fdata%2Ftanf%2FSection1&viewid=981949f0%2D98e1%2D4ace%2D8da8%2D436aca592a46

**Comment by ADPennington** [comment]
for the case number fix needed, can we explore using `isalpha()` validator instead of `isalnum()`

**Comment by ADPennington** [comment]
> for the case number fix needed, can we explore using `isalpha()` validator instead of `isalnum()`

moved this to #2794

**Comment by ADPennington** [comment]
capturing sync with @ysong001 on 3/5:

- we need to add back value 08 as valid possible value for item 49 (WPS) for 2-par calculation.

**Comment by ADPennington** [comment]
> capturing sync with @ysong001 on 3/5:
> 
> * we need to add back value 08 as valid possible value for item 49 (WPS) for 2-par calculation.

per today's DIGIT meeting we decided not to implement this change. cc: @ysong001 @elipe17

**Comment by elipe17** [comment]
> > capturing sync with @ysong001 on 3/5:
> > 
> > * we need to add back value 08 as valid possible value for item 49 (WPS) for 2-par calculation.
> 
> per today's DIGIT meeting we decided not to implement this change. cc: @ysong001 @elipe17

@ADPennington copy that. Thank you for clarifying. I have updated the PR with your other recommendations and re-requested your review.
